### PR TITLE
Add LRScheduler Composability E2E Tests

### PR DESCRIPTION
**Pull request by mlazos**
adds tests to verify the LRSchedulers correctly update the compiled optimizers without recompiles.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125653
* #125383
* #123753
* #123752
* #123751

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_dynamo/trace_rules.py [hunk 1]
```diff
@@ -3231,6 +3230,17 @@ def _module_dir(m: types.ModuleType):
...
+    "torch.optim.adam",
+    "torch.optim.adamw",
+    "torch.optim.radam",
+    "torch.optim.nadam",
+    "torch.optim.adamax",
+    "torch.optim.rmsprop",
+    "torch.optim.rprop",
+    "torch.optim.asgd",
+    "torch.optim.sgd",
+    "torch.optim.adadelta",
```
Comment (yanboliang) [inline torch/_dynamo/trace_rules.py:3243]: Is there any files under ```torch.optim``` which we don't want to inline? If no, we can just put one line here.

Comment (mlazos) [reply torch/_dynamo/trace_rules.py:3243]: Yeah I don't want to inline the LR schedulers (that's why I did this haha)

Although weirdly it looks like it removed a graph break somehow, I'm going to look into this tomorrow.

**Review by yanboliang — APPROVED** [review]

**Comment by mlazos** [comment]
> ## Merge failed
> **Reason**: 1 mandatory check(s) failed. The first few are:
> 
> * [pull / linux-focal-cuda12.1-py3.10-gcc9 / test (default, 4, 5, linux.4xlarge.nvidia.gpu)](https://github.com/pytorch/pytorch/actions/runs/9007308990/job/24747760487)
> 
> Dig deeper by [viewing the failures on hud](https://hud.pytorch.org/pytorch/pytorch/commit/7a471e939debb2b4bbe2b5fbb52567626e52d6c9)
> 
> Details for Dev Infra team

Test failure is happening elsewhere too
https://hud.pytorch.org/failure?name=pull%20%2F%20linux-focal-cuda12.1-py3.10-gcc9%20%2F%20test%20(default%2C%203%2C%205%2C%20linux.4xlarge.nvidia.gpu)&jobName=undefined&failureCaptures=%5B%22test_foreach.py%3A%3ATestForeachCUDA%3A%3Atest_binary_op_tensors_on_different_devices__foreach_div_cuda_bool%22%5D

**Pull request closed without merging**